cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -1104,6 +1375,90 @@ const FunctionDoc assume_timezone_doc{
     {"timestamps"},
     "AssumeTimezoneOptions"};
 
+const FunctionDoc years_between_doc{
+    "Compute the number of year boundaries between two timestamps",
```

Comment:
How about this?

```
const FunctionDoc years_between_doc{
    "Compute the number of years between two timestamps",
    ("Returns the number of year boundaries crossed from the first timestamp to the\n"
     "second. That is, the difference is calculated as if the timestamps were\n"
     "truncated to the year.\n"
     "Null values emit null."),
    {"start", "end"}};
```

Reply:
Better. How about using 'complete years passed/elapsed/..' instead of 'year boundaries crossed'?

Reply:
I suppose the reason why I'm using 'boundaries' is because the difference between 1998-12-01 and 1999-01-1 is '1 year' here even though a complete year hasn't passed.

Reply:
Oh, I thought the entire year needs to pass. Right, boundaries are better yeah.